### PR TITLE
Refactors creating and previewing deliveries

### PR DESCRIPTION
**Pull request by dleehr**
- Removes queryParam binding from the URLs in creating deliveries
- Changes deliveries.new route to create a delivery as its model, and sub-routes receive it as `delivery`
- Moves delivery previewing to a component
- Updates project and user picker to send actions based on selectedItems array
- Refactors delivery builder routes to be more declarative and uses a base class for error handling and computed properties
- Fixes issue in development mode where logging in would not redirect from login page

### CODE REVIEW

File: app/components/delivery-preview.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  },
+
+  generatePreview() {
+    this.set('emailMessage', null); // causes loading indicator to appear
+    // Delivery may be a model object that's not yet loaded (e.g. a relationship)
+    let delivery = this.get('delivery');
+    if(delivery.get('isLoaded')) {
+      delivery = Ember.RSVP.resolve(delivery);
+    }
+    delivery.then((loadedDelivery) => {
```
Comment (johnbradley) [inline app/components/delivery-preview.js]: Had to read this a couple times to figure out what is going on: 
If delivery is not a promise make it one.

**Comment by johnbradley** [comment]
Confirm delivery/preview page error if user clicks refresh:
screen shot 2018-08-31 at 9 01 39 am

File: app/components/delivery-preview.js (hunk 1 above)
Comment (dleehr) [reply app/components/delivery-preview.js]: I updated the comment around this. I still don't like how this works but have been digging in to understand why it's not working.

The underlying issue is that when I create a`{{delivery-preview}}` component in `resend-confirm.hbs`, the `delivery` object I provide is `model.delivery`, where `model` is a `duke-ds-project-transfer`.

When `delivery-preivew.js` accesses that `delivery`, it's a `DS.PromiseObject`. `DS.PromiseObject`s do not a have a `preview()` method, so the promise needs to be resolved to a "real" delivery before calling `preview()`. If I call `delivery.get('preview')()`, it does seem find the function, but then it's not bound to the right `this`.

Using `{{delivery-preview}}` from `deliveries/new/confirm.hbs` doesn't have this problem. That `delivery` object is the model for the route, so by the time it gets to the template, it's fully loaded and resolved.

Perhaps the best (and conventional) solution is for the `deliveries/show/confirm-resend` route to return the `delivery` as its model. The transfer can be accessed as a model from `deliveries/show`

File: app/controllers/deliveries/new/base.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    assert('Controller requires nextRoute property', this.get('nextRoute'));
+  },
+
+  /* Must override */
+  backRoute: null,
+  nextRoute: null,
+
+  /* May override */
+  disableNext: false,
+  working: false
```
[the remark this answers is not in the document]
Comment (dleehr) [reply app/controllers/deliveries/new/base.js]: No, not anymore. I experimented using it to show a loading indicator when fetching project permissions, but didn't have an obvious place to put that indicator.

**Comment by dleehr** [comment]
Fixes #19

**Pull request merged**